### DISCUSSION TITLE
Quality is not good even with thick model

### DISCUSSION THREAD

**Discussion by adityapatadia** [thekevinscott/UpscalerJS#1272] [Q&A]
We have used thick model variant on nodejs and the quality is still not very good compared to some online free upscalers.

Here are examples:
Original:
![Original](https://github.com/thekevinscott/UpscalerJS/assets/1086617/66e10905-3a9d-4dc4-a6b0-8861142f47d3)

ESRGAN-THICK-2x:
![esrgan-thick](https://github.com/thekevinscott/UpscalerJS/assets/1086617/505f7630-fddc-467c-a60b-0d85fa69e3a8)

Online upscaler 2x (using: https://www.iloveimg.com/upscale-image):
![online-upscaled](https://github.com/thekevinscott/UpscalerJS/assets/1086617/3cba1590-ecab-4d6c-be07-5b213f7773ef)

**Comment by adityapatadia** [comment]
What are models which can be used to achieve higher quality?

**Reply by thekevinscott** [reply]
It's not clear to me what model the linked website is using.

`esrgan-thick` is the highest quality model. [It's based off this paper from 2018](https://arxiv.org/abs/1809.00219), so it's not surprising that there are better quality models available today in 2023.

If there is a particular model you're interested in exploring, please link it and I'll investigate adding support. Not all models easily convert to Javascript, but I'd be happy to investigate if it's doable.

**Comment by adityapatadia** [comment]
I think real-esrgan seems much better model at task. Anyway it’s not urgent for us as we have deployed it in Python.—Regards,Aditya PatadiaOn 25-Nov-2023, at 8:37 PM, Kevin Scott ***@***.***> wrote:﻿
It's not clear to me what model the linked website is using.
esrgan-thick is the highest quality model. It's based off this paper from 2018, so it's not surprising that there are better quality models available today in 2023.
If there is a particular model you're interested in exploring, please link it and I'll investigate adding support. Not all models easily convert to Javascript, but I'd be happy to investigate if it's doable.
